syntax = "proto3";

package com.uiltiyai.guidance;

// Send a guidance program to the guidance service.
message GuidanceRequest {
  string program = 1;
}

// A single token from running a guidance program.
message GuidanceResponse {
  string text = 1;
}

// A tiny little frontend for guidance.
service Guidance {
  // Run a guidance program.
  rpc Guide(GuidanceRequest) returns (stream GuidanceResponse) {}
}

message HealthCheckRequest {
  string service = 1;
}

message HealthCheckResponse {
  enum ServingStatus {
    UNKNOWN = 0;
    SERVING = 1;
    NOT_SERVING = 2;
    SERVICE_UNKNOWN = 3;  // Used only by the Watch method.
  }
  ServingStatus status = 1;
}

service Health {
  rpc Check(HealthCheckRequest) returns (HealthCheckResponse);

  rpc Watch(HealthCheckRequest) returns (stream HealthCheckResponse);
}